### DISCUSSION TITLE
DWriteCore and Direct2d

### DISCUSSION THREAD

**Discussion by miquik** [microsoft/WindowsAppSDK#1651] [Q&A]
Hi all,
I’m just start studying and try to understand all directx technologies, so please excuse me if my question is stupid. If I understand correctly you suggest to use new DWriteCore interface to deal with text rendering in windows app; anyway you also said “ Direct2D and DWriteCore don't interoperate”, so my question is: what should I do if I want to render both Text and shapes? Need to use direct2d/dwrite?

thanks

**Comment by supthos** [comment]
Man. I feel you. It's like "who cares about text?" This is something I've asked myself. I'm trying to render custom fonts with DWriteCore for a while, and haven't succeeded. It's sad these questions go about being ignored for so long!

**Reply by castorix** [reply]
> Man. I feel you. It's like "who cares about text?" This is something I've asked myself. I'm trying to render custom fonts with DWriteCore for a while, and haven't succeeded. It's sad these questions go about being ignored for so long!

A bit late... but I updated an old sample ([WinUI3_SwapChainPanel_DWriteCore](https://github.com/castorix/WinUI3_SwapChainPanel_DWriteCore)) (DWriteCore or DirectWrite with Direct2D inside a SwapChainPanel with basic animations), to be improved...
